src/operations/rotate.rs

```diff
@@ -0,0 +1,129 @@
+//! Image rotation operations.
+
+use image::{DynamicImage, ImageBuffer, Pixel, Primitive};
+use imageproc::{definitions::Clamp, geometric_transformations::Interpolation};
+
+use crate::{arg_parsers::RotateGeometry, error::MagickError, image::Image};
+
+fn white_pixel<P>() -> P
+where
+    P: Pixel + Send + Sync,
+    P::Subpixel: Send + Sync + Primitive,
+{
+    let max_value = P::Subpixel::DEFAULT_MAX_VALUE;
+    let channels = P::CHANNEL_COUNT as usize;
+
+    // Create a vector with all channels set to maximum value
+    let subpixels = vec![max_value; channels];
+    P::from_slice(&subpixels).clone()
+}
+
+fn rotate_inner<P>(
+    image: &mut ImageBuffer<P, Vec<P::Subpixel>>,
+    theta: f32,
+) -> Result<(), MagickError>
+where
+    P: 'static + Pixel + Send + Sync,
+    P::Subpixel: Send + Sync + Primitive + Into<f32> + Clamp<f32>,
+{
+    let width = image.width();
+    let height = image.height();
+
+    let (sin_theta, cos_theta) = theta.sin_cos();
+
+    let dst_width = (width as f32 * cos_theta + height as f32 * sin_theta).ceil() as u32;
+    let dst_height = (width as f32 * sin_theta + height as f32 * cos_theta).ceil() as u32;
```

Comment:
I tend to use f64 just on general principle to avoid running into precision issues. Is `f32` here to replicate imagemagick behavior? If not I'd prefer f64, this really isn't a hot path so I don't think there's anything to be gained from f32.